versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/BaseMergeTransplantSquash.java

```diff
@@ -132,9 +139,23 @@ private CreateCommit createSquashCommit(
       commitBuilder.addSecondaryParents(mergeFromId);
     }
 
+    Predicate<StoreKey> filter =
```

Comment:
This filter is the central piece of this PR - it removes all `DROP`ped keys from the diff.